tensorflow/compiler/xla/service/gpu/gemm_rewriter.cc

```diff
@@ -187,18 +203,55 @@ class GemmRewriterVisitor : public DfsHloRewriteVisitor {
   }
 
   Status HandleAdd(HloInstruction *instr) override {
-    HloInstruction *bias, *existing_gemm;
+    HloInstruction *bias, *existing_gemm, *slice;
 
-    // First, try to match vector bias add, so we might elide the broadcast.
+    // Attempt to elide broadcast and fuse addition of a vector bias into GEMM.
```

Comment:
Some of these may be existing bugs (sorry), but where are we checking that all of these ops have exactly one use?  Are we sure this is correct?  It might be worth adding tests.

Reply:
You're right that the number of users should be checked. I've added WithOneUser() constraints to the patterns.

Reply:
I added tests for GEMM and Slice with multiple users. Bitcast is a bit complicated since manually inserted instructions cause this error: https://github.com/tensorflow/tensorflow/blob/8508b747acb287898651274d8ee3e800f0a13824/tensorflow/compiler/xla/service/layout_assignment.cc#L1963